qualtran/bloqs/chemistry/trotter/trotterized_unitary.py

```diff
@@ -0,0 +1,121 @@
+#  Copyright 2023 Google LLC
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      https://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+"""Bloq for building a Trotterized unitary"""
+
+from functools import cached_property
+from typing import Dict, Iterable, Union
+
+import attrs
+import sympy
+
+from qualtran import Bloq, bloq_example, BloqBuilder, BloqDocSpec, Signature, Soquet, SoquetT
+
+
+@attrs.frozen
+class TrotterizedUnitary(Bloq):
```

Comment:
I think you'll have to add `TrotterizedUnitary` to the `RESOLVER_DICT`; hopefully the tests will fail since you run `bloq_autotester` and it should automatically test for serialization and raise an error.

Reply:
Is there an automatic way to build up this dictionary?

Reply:
I used the code in `dev_tools/bloq_finder.py` to `get_bloq_classes` and copy/paste to construct the dictionary; so I'm sure it can be done. The only issue is that this code is not part of the qualtran package and uses calls to `git` to get all files and recursively extract class names. 

We could add a decorator that "registers" the classes with the serialization infrastructure and therefore adds the class to the resolver dictionary. Then, we'll have to decorate all bloqs with this decorator. I'll have to think more about other potential ways to achieve this.